clang/tools/archgen-mlir/ArchGenMLIR.cpp

```diff
@@ -0,0 +1,972 @@
+//===- ArchGenMLIR.cpp ---------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Hybrid (LLVMIR, MLIR) IR-gen used for the compiler implementation based on
+// archgenlib
+// The Compiler part of ArchGenMLIR is contained inside a clang plugin. clang
+// doesn't support the plugin behavior I needed so I slightly modified clang to
+// get the behavior i needed.
```

Comment:
```suggestion
// doesn't support the plugin behavior needed so clang has been slightly modified to
// get the needed behavior.
```
Since this is a strong move, at least you should precise what was the problem?